scripts/microbe_stage/microbe_operations.as

```diff
@@ -895,22 +895,63 @@ bool validPlacement(CellStageWorld@ world, ObjectID microbeEntity, const Organel
     return touching;
 }
 
-// speciesName decides the template to use, while individualName is
-// used for referencing the instance
-ObjectID spawnMicrobe(CellStageWorld@ world, Float3 pos, const string &in speciesName,
-    bool aiControlled)
+// function creates a microbe to be spawned by either the spawnMicrobe function or
+// the spawnBacteria function, reduces code duplication
+ObjectID createMicrobe(CellStageWorld@ world, Float3 pos, const string &in speciesName,
+    bool aiControlled, string typeOfMicrobe)
 {
     assert(world !is null);
     assert(speciesName != "");
 
     if(pos.Y != 0)
-        LOG_WARNING("spawnMicrobe: spawning at y-coordinate: " + pos.Y);
+        LOG_WARNING("spawn" + typeOfMicrobe + ": spawning at y-coordinate: " + pos.Y);
+
+    auto processor = getProcessorComponent(world, speciesName);
 
+    if(processor is null){
+        LOG_ERROR("Skipping " + typeOfMicrobe + " spawn because species '" + speciesName +
+            "' doesn't have a processor component");
 
+        return NULL_OBJECT;
+    }
+    
+    return _createMicrobeEntity(world, aiControlled, speciesName, false);
+}
+
+// speciesName decides the template to use, while individualName is
+// used for referencing the instance
+ObjectID spawnMicrobe(CellStageWorld@ world, Float3 pos, const string &in speciesName,
+    bool aiControlled)
+{
     // Create microbeEntity with correct template, physics and species name
     auto microbeEntity = _createMicrobeEntity(world, aiControlled, speciesName,
+
+    auto processor = getProcessorComponent(world, speciesName);
+
+    if(processor is null){
+        LOG_ERROR("Skipping " + typeOfMicrobe + " spawn because species '" + speciesName +
+            "' doesn't have a processor component");
+
+        return NULL_OBJECT;
+    }
+    
+    return _createMicrobeEntity(world, aiControlled, speciesName,
         // in_editor
         false);
+}
+
+// speciesName decides the template to use, while individualName is
+// used for referencing the instance
+ObjectID spawnMicrobe(CellStageWorld@ world, Float3 pos, const string &in speciesName,
```

Comment:
I think that this should be the *only* microbe spawning function, and it should automatically based on the species determine if this is a bacteria or not.

Reply:
Alright, thanks for the feedback guys, you guys type really fast. I'll make the changes.